sdk/core/amqp-common/src/auth/aad.ts

```diff
@@ -38,25 +30,10 @@ export class AadTokenProvider implements TokenProvider {
    */
   readonly tokenValidTimeInSeconds: number = 3599;
 
-  constructor(
-    credentials:
-      | ApplicationTokenCredentials
-      | UserTokenCredentials
-      | DeviceTokenCredentials
-      | MSITokenCredentials
-  ) {
-    if (
-      !credentials ||
-      (credentials &&
-        !(
-          credentials instanceof ApplicationTokenCredentials ||
-          credentials instanceof UserTokenCredentials ||
-          credentials instanceof DeviceTokenCredentials ||
-          credentials instanceof MSITokenCredentials
-        ))
-    ) {
+  constructor(credentials: TokenCredentialInterface) {
+    if (!credentials) {
       throw new TypeError(
-        "'credentials' is a required parameter and must be an instance of " +
+        "'credentials' is a required parameter and must be of type compatible with " +
```

Comment:
Pedantically, there are two notions of type compatibility in TypeScript - subtype compatibility and assignment compatibility. What's meant here is the latter. I would word this as: "must be assignable to the type TokenCredentials", or even more directly, "must implement a getToken() method".